// @nodegui/nodegui/dist/index.js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
// bootstrap
require("./lib/core/bootstrap");
// Enums:
__export(require("./lib/QtEnums"));
// Gui:
var QApplication_1 = require("./lib/QtGui/QApplication");
exports.QApplication = QApplication_1.QApplication;
var QKeySequence_1 = require("./lib/QtGui/QKeySequence");
exports.QKeySequence = QKeySequence_1.QKeySequence;
var QPixmap_1 = require("./lib/QtGui/QPixmap");
exports.QPixmap = QPixmap_1.QPixmap;
var QIcon_1 = require("./lib/QtGui/QIcon");
exports.QIcon = QIcon_1.QIcon;
exports.QIconMode = QIcon_1.QIconMode;
exports.QIconState = QIcon_1.QIconState;
var QCursor_1 = require("./lib/QtGui/QCursor");
exports.QCursor = QCursor_1.QCursor;
var QTextOption_1 = require("./lib/QtGui/QTextOption");
exports.QTextOptionWrapMode = QTextOption_1.QTextOptionWrapMode;
var QClipboard_1 = require("./lib/QtGui/QClipboard");
exports.QClipboard = QClipboard_1.QClipboard;
exports.QClipboardMode = QClipboard_1.QClipboardMode;
// Events: Maybe a separate module ?
var QKeyEvent_1 = require("./lib/QtGui/QEvent/QKeyEvent");
exports.QKeyEvent = QKeyEvent_1.QKeyEvent;
var EventWidget_1 = require("./lib/core/EventWidget");
exports.BaseWidgetEvents = EventWidget_1.BaseWidgetEvents;
// Abstract:
var QWidget_1 = require("./lib/QtWidgets/QWidget");
exports.NodeWidget = QWidget_1.NodeWidget;
var QLayout_1 = require("./lib/QtWidgets/QLayout");
exports.NodeLayout = QLayout_1.NodeLayout;
var QAbstractScrollArea_1 = require("./lib/QtWidgets/QAbstractScrollArea");
exports.QAbstractScrollArea = QAbstractScrollArea_1.QAbstractScrollArea;
var QAbstractSlider_1 = require("./lib/QtWidgets/QAbstractSlider");
exports.QAbstractSlider = QAbstractSlider_1.QAbstractSlider;
// Widgets:
var QWidget_2 = require("./lib/QtWidgets/QWidget");
exports.QWidget = QWidget_2.QWidget;
exports.QWidgetEvents = QWidget_2.QWidgetEvents;
var QCheckBox_1 = require("./lib/QtWidgets/QCheckBox");
exports.QCheckBox = QCheckBox_1.QCheckBox;
exports.QCheckBoxEvents = QCheckBox_1.QCheckBoxEvents;
var QLabel_1 = require("./lib/QtWidgets/QLabel");
exports.QLabel = QLabel_1.QLabel;
exports.QLabelEvents = QLabel_1.QLabelEvents;
var QDial_1 = require("./lib/QtWidgets/QDial");
exports.QDial = QDial_1.QDial;
exports.QDialEvents = QDial_1.QDialEvents;
var QLineEdit_1 = require("./lib/QtWidgets/QLineEdit");
exports.QLineEdit = QLineEdit_1.QLineEdit;
exports.QLineEditEvents = QLineEdit_1.QLineEditEvents;
var QMainWindow_1 = require("./lib/QtWidgets/QMainWindow");
exports.QMainWindow = QMainWindow_1.QMainWindow;
exports.QMainWindowEvents = QMainWindow_1.QMainWindowEvents;
var QProgressBar_1 = require("./lib/QtWidgets/QProgressBar");
exports.QProgressBar = QProgressBar_1.QProgressBar;
exports.QProgressBarEvents = QProgressBar_1.QProgressBarEvents;
var QPushButton_1 = require("./lib/QtWidgets/QPushButton");
exports.QPushButton = QPushButton_1.QPushButton;
exports.QPushButtonEvents = QPushButton_1.QPushButtonEvents;
var QSpinBox_1 = require("./lib/QtWidgets/QSpinBox");
exports.QSpinBox = QSpinBox_1.QSpinBox;
exports.QSpinBoxEvents = QSpinBox_1.QSpinBoxEvents;
var QRadioButton_1 = require("./lib/QtWidgets/QRadioButton");
exports.QRadioButton = QRadioButton_1.QRadioButton;
exports.QRadioButtonEvents = QRadioButton_1.QRadioButtonEvents;
var QTabWidget_1 = require("./lib/QtWidgets/QTabWidget");
exports.QTabWidget = QTabWidget_1.QTabWidget;
exports.QTabWidgetEvents = QTabWidget_1.QTabWidgetEvents;
var QMenu_1 = require("./lib/QtWidgets/QMenu");
exports.QMenu = QMenu_1.QMenu;
exports.QMenuEvents = QMenu_1.QMenuEvents;
var QMenuBar_1 = require("./lib/QtWidgets/QMenuBar");
exports.QMenuBar = QMenuBar_1.QMenuBar;
exports.QMenuBarEvents = QMenuBar_1.QMenuBarEvents;
var QPlainTextEdit_1 = require("./lib/QtWidgets/QPlainTextEdit");
exports.QPlainTextEdit = QPlainTextEdit_1.QPlainTextEdit;
exports.QPlainTextEditEvents = QPlainTextEdit_1.QPlainTextEditEvents;
exports.LineWrapMode = QPlainTextEdit_1.LineWrapMode;
var QScrollArea_1 = require("./lib/QtWidgets/QScrollArea");
exports.QScrollArea = QScrollArea_1.QScrollArea;
exports.QScrollAreaEvents = QScrollArea_1.QScrollAreaEvents;
var QSystemTrayIcon_1 = require("./lib/QtWidgets/QSystemTrayIcon");
exports.QSystemTrayIcon = QSystemTrayIcon_1.QSystemTrayIcon;
exports.QSystemTrayIconEvents = QSystemTrayIcon_1.QSystemTrayIconEvents;
var QAction_1 = require("./lib/QtWidgets/QAction");
exports.QAction = QAction_1.QAction;
exports.QActionEvents = QAction_1.QActionEvents;
var QShortcut_1 = require("./lib/QtWidgets/QShortcut");
exports.QShortcut = QShortcut_1.QShortcut;
exports.QShortcutEvents = QShortcut_1.QShortcutEvents;
// Core
var QObject_1 = require("./lib/QtCore/QObject");
exports.QObject = QObject_1.QObject;
exports.NodeObject = QObject_1.NodeObject;
// Layouts:
var QGridLayout_1 = require("./lib/QtWidgets/QGridLayout");
exports.QGridLayout = QGridLayout_1.QGridLayout;
var FlexLayout_1 = require("./lib/core/FlexLayout");
exports.FlexLayout = FlexLayout_1.FlexLayout;
// Others:
var StyleSheet_1 = require("./lib/core/Style/StyleSheet");
exports.StyleSheet = StyleSheet_1.StyleSheet;
var Component_1 = require("./lib/core/Component");
exports.Component = Component_1.Component;
