// @nodegui/nodegui/dist/lib/QtGui/QClipboard.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Component_1 = require("../core/Component");
const helpers_1 = require("../utils/helpers");
class QClipboard extends Component_1.Component {
    constructor(native) {
        super();
        if (helpers_1.checkIfNativeElement(native)) {
            this.native = native;
        }
        else {
            throw new Error('QClipboard cannot be initialised this way. Use QApplication::clipboard()');
        }
    }
    clear(mode) {
        this.native.clear(mode);
    }
    setText(text, mode) {
        this.native.setText(text, mode);
    }
    text(mode) {
        return this.native.text(mode);
    }
}
exports.QClipboard = QClipboard;
var QClipboardMode;
(function (QClipboardMode) {
    QClipboardMode[QClipboardMode["Clipboard"] = 0] = "Clipboard";
    QClipboardMode[QClipboardMode["Selection"] = 1] = "Selection";
    QClipboardMode[QClipboardMode["FindBuffer"] = 2] = "FindBuffer";
})(QClipboardMode = exports.QClipboardMode || (exports.QClipboardMode = {}));
