// @nodegui/nodegui/dist/lib/QtWidgets/QCheckBox.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const addon_1 = __importDefault(require("../utils/addon"));
const QWidget_1 = require("./QWidget");
const EventWidget_1 = require("../core/EventWidget");
exports.QCheckBoxEvents = Object.freeze(Object.assign(Object.assign({}, EventWidget_1.BaseWidgetEvents), { toggled: 'toggled' }));
class QCheckBox extends QWidget_1.NodeWidget {
    constructor(parent) {
        let native;
        if (parent) {
            native = new addon_1.default.QCheckBox(parent.native);
        }
        else {
            native = new addon_1.default.QCheckBox();
        }
        super(native);
        this.native = native;
        this.nodeParent = parent;
    }
    setText(text) {
        // react:✓ //TODO:getter
        this.native.setText(text);
    }
    setChecked(check) {
        // react:✓
        this.native.setChecked(check);
    }
    isChecked() {
        // react:✓
        return this.native.isChecked();
    }
}
exports.QCheckBox = QCheckBox;
