// @nodegui/nodegui/dist/lib/QtWidgets/QLabel.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const addon_1 = __importDefault(require("../utils/addon"));
const QWidget_1 = require("./QWidget");
const EventWidget_1 = require("../core/EventWidget");
exports.QLabelEvents = Object.freeze(Object.assign({}, EventWidget_1.BaseWidgetEvents));
class QLabel extends QWidget_1.NodeWidget {
    constructor(parent) {
        let native;
        if (parent) {
            native = new addon_1.default.QLabel(parent.native);
        }
        else {
            native = new addon_1.default.QLabel();
        }
        super(native);
        this.native = native;
        this.nodeParent = parent;
    }
    setWordWrap(on) {
        this.native.setWordWrap(on);
    }
    wordWrap() {
        return this.native.wordWrap();
    }
    setText(text) {
        this.native.setText(`${text}`);
    }
    text() {
        return this.native.text();
    }
    setPixmap(pixMap) {
        this.native.setPixmap(pixMap.native);
        this._pixmap = pixMap;
    }
    pixmap() {
        return this._pixmap;
    }
    setOpenExternalLinks(open) {
        this.native.setOpenExternalLinks(open);
    }
}
exports.QLabel = QLabel;
