// @nodegui/nodegui/dist/lib/QtWidgets/QPushButton.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const addon_1 = __importDefault(require("../utils/addon"));
const QWidget_1 = require("./QWidget");
const EventWidget_1 = require("../core/EventWidget");
exports.QPushButtonEvents = Object.freeze(Object.assign(Object.assign({}, EventWidget_1.BaseWidgetEvents), { clicked: 'clicked', pressed: 'pressed', released: 'released', toggled: 'toggled' }));
class QPushButton extends QWidget_1.NodeWidget {
    constructor(parent) {
        let native;
        if (parent) {
            native = new addon_1.default.QPushButton(parent.native);
        }
        else {
            native = new addon_1.default.QPushButton();
        }
        super(native);
        this.nodeParent = parent;
        this.native = native;
    }
    setText(text) {
        // react:✓, //TODO:getter
        this.native.setText(`${text}`);
    }
    setFlat(isFlat) {
        // react:✓, //TODO:getter
        this.native.setFlat(isFlat);
    }
    setIcon(icon) {
        // react:✓, //TODO:getter
        this.native.setIcon(icon.native);
    }
}
exports.QPushButton = QPushButton;
