// @nodegui/nodegui/dist/lib/QtWidgets/QScrollArea.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const addon_1 = __importDefault(require("../utils/addon"));
const EventWidget_1 = require("../core/EventWidget");
const QAbstractScrollArea_1 = require("./QAbstractScrollArea");
exports.QScrollAreaEvents = Object.freeze(Object.assign({}, EventWidget_1.BaseWidgetEvents));
class QScrollArea extends QAbstractScrollArea_1.QAbstractScrollArea {
    constructor(parent) {
        let native;
        if (parent) {
            native = new addon_1.default.QScrollArea(parent.native);
        }
        else {
            native = new addon_1.default.QScrollArea();
        }
        super(native);
        this.native = native;
        this.nodeParent = parent;
    }
    setWidget(widget) {
        // react:✓, //TODO:getter
        this.contentWidget = widget;
        this.native.setWidget(widget.native);
    }
    takeWidget() {
        // react:✓
        const contentWidget = this.contentWidget;
        this.contentWidget = null;
        if (contentWidget) {
            this.native.takeWidget();
            return contentWidget;
        }
        return null;
    }
    setWidgetResizable(resizable) {
        this.native.setWidgetResizable(resizable);
    }
    widgetResizable() {
        return this.native.widgetResizable();
    }
}
exports.QScrollArea = QScrollArea;
